From 8e61a09aeb26bfef3333694ffcef8656541a7831 Mon Sep 17 00:00:00 2001
From: Abu Kashem <akashem@redhat.com>
Date: Fri, 29 Apr 2022 17:51:54 -0400
Subject: [PATCH] UPSTREAM: <carry>: fix [sig-auth] ServiceAccounts no
 secret-based service account token should be auto-generated

OpenShift-Rebase-Source: a03143879c0
---
 test/e2e/auth/service_accounts.go | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/e2e/auth/service_accounts.go b/test/e2e/auth/service_accounts.go
index 654600d7e32..7130e2c1c5c 100644
--- a/test/e2e/auth/service_accounts.go
+++ b/test/e2e/auth/service_accounts.go
@@ -64,7 +64,11 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 			time.Sleep(10 * time.Second)
 			sa, err := f.ClientSet.CoreV1().ServiceAccounts(f.Namespace.Name).Get(ctx, "default", metav1.GetOptions{})
 			framework.ExpectNoError(err)
-			gomega.Expect(sa.Secrets).To(gomega.BeEmpty())
+
+			// TODO: OpenShift creates a Secret with a prefix "default-dockercfg", is the
+			//  following assertion valid?
+			gomega.Expect(sa.Secrets).To(gomega.HaveLen(1))
+			gomega.Expect(sa.Secrets[0].Name).To(gomega.HavePrefix("default-dockercfg"))
 		}
 	})
 
-- 
2.43.0

